File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/DocumentService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.DocumentDTO;
import com.hexaware.automobileinsurance.model.Document;

public interface DocumentService {

    Document uploadDocument(DocumentDTO dto);

    Document getDocumentById(Integer documentId);

    List<Document> getAllDocuments();

    void deleteDocument(Integer documentId);
    
    
}

File: automobile-insurance-frontend/src/services/DocumentService.js
import api from "../api/api";

const DocumentService = {

    getAllDocuments() {

        return api.get("/documents");

    },

    getDocumentById(id) {

        return api.get(`/documents/${id}`);

    },

    uploadDocument(document) {

        return api.post("/documents", document);

    },

    deleteDocument(id) {

        return api.delete(`/documents/${id}`);

    }
    
};

export default DocumentService;